### PR TITLE
Add docker support for development

### PR DESCRIPTION
**Pull request by ftsell**
The dockerfile has been adapted to also produce a container which is ready for a full development setup if docker build is run to do so.
Additionally a script (`dev_up.sh`) has been added which simplifies getting a development environment up and running.

### CODE REVIEW

**Comment by ftsell** [comment]
@Flova the api key in the tests does not seem to work anymore

**Comment by Flova** [comment]
The API key is from Timon. I think he added an IP restriction, so that only the deployment server can call the API and forgot about the CI at GitHub.

**Comment by ftsell** [comment]
If noone is interest in this I can close the PR. I thought it was neat though

**Comment by Flova** [comment]
I think it is neat too, but I didnt had the resources to test this. Sorry

File: README.md [hunk 1]
```diff
@@ -103,10 +114,16 @@ npm run serve
...
 ## Configuration
-There are two main ways of configuring the application: Frontend and backend.
-While the defaults were chosen in such a way that they work in a development environment
-(the docker container has different [ones](./docker/settings.json)) they might need change in a production setup.
+### How to configure
+The application is configured at two places:
+1) Environment variables for the backend
+2) A `settings.json` file for the frontend
+
+Defaults defined in the source code were chosen to work in a development environment. The
```
Comment (JanGut) [inline README.md:122]: This sentence is a bit unclear. What do you meant by "defaults in source code"?

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+#       base           |
+#     /      \         |
+#     |   backend  frontend
+#     |         \   /
+#    dev        final
+#
+# base: necessary base runtime and build dependencies like python
+# backend: installed backend src-code
+# fronted: installed frontend src-code
+# dev: configuration links but no code since that gets mounted
```
Comment (JanGut) [inline Dockerfile:13]: How do I mount my source code? Or does this happen automatically?

Comment (ftsell) [reply Dockerfile:13]: It's still docker so mounts happen when the container starts. In this case, it happens here:
 https://github.com/TheNerdful8/Lecture2Gether/blob/92c91f67cce4b5229b47c88e4930b1ad108e3a7c/dev_up.sh#L12

File: README.md (hunk 1 above)
Comment (ftsell) [reply README.md:122]: Defaults like where backend API and SocketIO server are reachable.

File: README.md [hunk 3]
```diff
@@ -64,6 +64,17 @@ docker run -p 8000:8000 thenerdful8/lecture2gether
 This will start the whole application stack and expose it at
 [http://localhost:8000/](http://localhost:8000).
 
+
+### Full Development Environment
+For development we recommend building a dev container (defined in default Dockerfile) and
```
[the remark this answers is not in the document]
Comment (ftsell) [reply README.md:69]: No there is no other one. I actually wanted to make it clear that there is no separate Dockerfile for a development container but that it instead defined in the default Dockerfile as well.

**Comment by ftsell** [comment]
@JanGut 
> I couldn't test these changes successfully, but I have added a few questions.

Why not? What went wrong?

**Comment by JanGut** [comment]
TODO:

- [ ] Add `.venv` to gitignore

**Comment by ftsell** [comment]
@JanGut 

As you can see here, the folder should be named `.vens` and not `.venv`:
https://github.com/TheNerdful8/Lecture2Gether/blob/33d3ce90f519c55c8e0644d86aab57ed9de540ab/dev_up.sh#L14
And that folder should already be ignored from here:
https://github.com/TheNerdful8/Lecture2Gether/blob/33d3ce90f519c55c8e0644d86aab57ed9de540ab/.gitignore#L2

**Pull request merged**